### PR TITLE
[react-is] Add AsyncMode, ForwardRef and Profiler

### PR DESCRIPTION
**Pull request by christianchown**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.npmjs.com/package/react-is/v/16.5.2
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

*NOTE:* tests not included for `AsyncMode` and `Profiler` as these are not yet a part of @types/react

### CODE REVIEW

File: types/react-is/index.d.ts [hunk 1]
```diff
@@ -10,17 +11,22 @@ import { ReactElement, ReactType } from 'react';
 
 export function typeOf(value: any): symbol | undefined;
 export function isValidElementType(value: any): value is ReactType;
-
+export function isAsyncMode(value: any): value is ReactType;
```
Comment (AviVahl) [inline types/react-is/index.d.ts]: Could you clarify why the return type is not `ReactElement<any>`?
Looking at the docs, function returns `true` if value is a JSX element containing `React.unstable_AsyncMode` type.

**Review by AviVahl — CHANGES_REQUESTED** [review]

File: types/react-is/index.d.ts (hunk 1 above)
Comment (christianchown) [reply types/react-is/index.d.ts]: Apologies and nice catch. Have added a new commit addressing this.

**Review by AviVahl — APPROVED** [review]

**Comment by AviVahl** [comment]
Odd that the "revision needed" label was not auto-removed (by bot) once I changed review status to "approved".

**Comment by christianchown** [comment]
There may be a way to mark the review at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29133/files/c41fd250fa3f8f5b16c40b2ee838109aeea45509 as stale, having been superceded by 29ebc13c7c268d049721428358b2f3d0c5eae1a6, but I can't see it

**Pull request merged**